aplt/runner.py

```diff
@@ -0,0 +1,153 @@
+"""Scenario Runner"""
+import importlib
+import sys
+from collections import deque
+
+import treq
+from autobahn.twisted.websocket import (
+    connectWS,
+    WebSocketClientFactory
+)
+from docopt import docopt
+from twisted.internet import reactor, ssl, task
+from twisted.python import log
+
+from aplt import __version__
+from aplt.client import (
+    CommandProcessor,
+    WSClientProtocol
+)
+
+
+class RunnerHarness(object):
+    def __init__(self, websocket_url):
+        self._factory = WebSocketClientFactory(
+            websocket_url,
+            headers={"Origin": "localhost:9000"},
+            debug=False)
+        self._factory.protocol = WSClientProtocol
+        self._factory.protocol.harness = self
+        if websocket_url.startswith("wss"):
+            self._factory_context = ssl.ClientContextFactory()
+        else:
+            self._factory_context = None
+
+        self._crypto_key = """\
+keyid="http://example.org/bob/keys/123;salt="XZwpw6o37R-6qoZjw6KwAw"\
+"""
+
+        # Processor and Websocket client vars
+        self._scenarios = {}
+        self._processors = {}
+        self._ws_clients = {}
+        self._connect_waiters = deque()
+
+    def register_scenario(self, name, scenario):
+        """Register a scenario to run"""
+        self._scenarios[name] = scenario
+
+    def run(self):
+        """Start all registered scenarios and the twisted event loop"""
+        for scenario in self._scenarios.values():
+            # Create the processor and start it
+            processor = CommandProcessor(scenario, self)
+            processor.run()
+            self._processors[processor] = True
+
+    def connect(self, processor):
+        """Start a connection for a processor and queue it for when the
+        connection is available"""
+        connectWS(self._factory, contextFactory=self._factory_context)
+        self._connect_waiters.append(processor)
+
+    def send_notification(self, processor, url, data, ttl):
+        """Send out a notification to a url for a processor"""
+        url = url.encode("utf-8")
+        if data:
+            d = treq.post(
+                url,
+                data,
+                headers={
+                    "Content-Type": "application/octet-stream",
+                    "Content-Encoding": "aesgcm-128",
+                    "Encryption": self._crypto_key,
+                    "Encryption-Key":
+                        'keyid="a1"; key="JcqK-OLkJZlJ3sJJWstJCA"',
```

Comment:
The key here is just meaningless, right? Tempted to suggest using something to reflect that like "Invalid-Key-Used-Here".